### DISCUSSION TITLE
Support embedded Cloud CodeSandBox

### DISCUSSION THREAD

**Discussion by MantisClone** [GitbookIO/community#211] [Feature Requests]
### What's your feature/idea?

Today, Gitbook supports embedding a [Browser CodeSandBox](https://codesandbox.io/docs/learn/sandboxes/overview?tab=browser). I would like it to also support embedding a [Cloud CodeSandBox](https://codesandbox.io/docs/learn/sandboxes/overview?tab=cloud). 

Here's an image showing an embedded Browser CodeSandBox side-by-side with an embedded Cloud CodeSandBox.
![image](https://github.com/GitbookIO/community/assets/2530913/bc95795b-638b-4b5c-88da-3fe7c8078f90)

Example Browser CodeSandBox URL: https://codesandbox.io/embed/retrieve-a-users-requests-pwe9o2?codemirror=1&fontsize=12&hidenavigation=1&theme=light&view=split

Example Cloud CodeSandBox URL: https://codesandbox.io/p/sandbox/request-network-retreive-a-users-requests-zsop8n?embed=1

### What's your desired outcome?

I would like the embedded Cloud CodeSandBox to render an interactive editor and app view similar to the embedded Browser CodeSandBox. An example of an embedded Cloud CodeSandBox can be seen in this recent [CodeSandBox Blog post](https://codesandbox.io/blog/unique-database-for-every-pr)

![image](https://github.com/GitbookIO/community/assets/2530913/38a6928e-2115-4ff9-981e-1157ce9d6c6b)

### What's the impact of this for you?

- [ ] Critical
- [X] High
- [ ] Medium
- [ ] Low

### Anything else?

I don't know if this feature needs to be implemented on the Gitbook side or on the CodeSandBox side.

This is a high impact on my work, and the lack of Cloud CodeSandBox support could mean I switch to a different documentation generator or end up developing my own custom documentation site.

Related #148

**Comment by addisonschultz** [comment]
Great request - we're definitely looking into this!

**Comment by MantisClone** [comment]
@addisonschultz Any updates on this?

**Reply by addisonschultz** [reply]
We're still exploring a possible [SandPack](https://sandpack.codesandbox.io/) integration! But this being said, I don't have an accurate timeline for you at the moment—We've got a few things to review functionally to see if this will work. 

Stay tuned!

**Reply by tromika** [reply]
@addisonschultz I'm Tamas from the CodeSandbox team. I don't think this requires a sandpack integration. It's more about our embeds https://codesandbox.io/embed/retrieve-a-users-requests-pwe9o2?codemirror=1&fontsize=12&hidenavigation=1&theme=light&view=split rendered in an iframe in Gitbook but the new sandboxes https://codesandbox.io/p/sandbox/request-network-retreive-a-users-requests-zsop8n?embed=1 are not.
I wonder if the problem is on our side so happy to ship improvements if needed.

**Reply by addisonschultz** [reply]
Hi Tamas! Thanks for joining in the conversation. Long story short - we're using iFramely to handle the unfurling and display of links in GitBook pages. I'm noticing here the new cloud sandboxes are also not working: https://iframely.com/try?url=https%3A%2F%2Fcodesandbox.io%2Fp%2Fsandbox%2Frequest-network-retreive-a-users-requests-zsop8n%3Fembed%3D1

Is there anything you can do on your end to investigate that easily? 

I mentioned Sandpack as we're also looking into building an integration that allows you to natively craft IDEs in a GitBook page - which isn't an immediate workaround but could still offer some value.

If you're able to help look into the unfurling aspect in the new URLs, that would be a great starting point to help resolve this overall!
